src/Util/check_hls_status.php

```diff
@@ -27,29 +26,47 @@ function fetch(string $url): array
     ];
 }
 
-function parsePlaylist(string $ts): array
+function parsePlaylist(string $body, string $livestream_type): array
 {
     // process the string
-    $pieces = explode("\n", $ts); // make an array out of curl return value
+    $pieces = explode("\n", $body); // make an array out of curl return value
     $pieces = array_map('trim', $pieces); // remove unnecessary space
-    $chunklists = array_filter($pieces, function (string $piece): bool { // pluck out ts urls
-        return strtolower(substr($piece, -3)) === '.ts';
+
+    $chunklists = array_filter($pieces, function (string $piece) use ($livestream_type) { // pluck out ts urls
+        $is_valid_chunk = false;
+        // By default considering HLS, which has ['.m3u8', '.m3u']
+        $playlist_exts = ['.m3u8', '.m3u'];
+        if ($livestream_type === 'mpegts') {
+            $playlist_exts = ['.ts'];
+        }
+        foreach ($playlist_exts as $ext) {
+            $len = strlen($ext) ;
+            if (strtolower(substr($piece, -($len))) === $ext) {
+                $is_valid_chunk = true;
+                break;
+            }
+        }
+        return $is_valid_chunk;
     });
     return $chunklists;
 }
 
 $url = urldecode(filter_input(INPUT_GET, 'url'));
+$livestream_type = filter_input(INPUT_GET, 'livestream_type');
```

Comment:
As we spoke about using the wrapper, I am totally agree with you, and will apply.
However, I am afraid this `check_hls_status` is going to be an exception, because it is called by an ajax call from the player.js and has no access to $DIC or other ILIAS stuff.

Reply:
The only place to apply the wrapper (or in case of backward compatibility http & getQueryParams) which is related to this PR (i.e. player) is in `classes/Player/class.xoctPlayerGUI.php` and it has been covered in no. 5:  https://github.com/opencast-ilias/OpenCast/pull/204/commits/b9a38886023f22c5333b495f63bd3c39bb1c86c0